### PR TITLE
DUPLO-9973 Ensure cached AWS credentials are still valid with ping

### PR DESCRIPTION
**Pull request by duplodavid**
## Change description

This builds off of pre-requisite PR: https://github.com/duplocloud/duplo-jit/pull/21
Adding ping of AWS get-caller-identity API to ensure the cached credentials are still valid for a `duplo-jit aws` call separate from checking the cached expiration date.

I also factored out and reused the code that resolves the tenant ID and tenant name in the k8s case. Although not essential here, it removes some duplicate logic and makes the cache key scheme consistent with the improved scheme used for the k8s cached credentials when the tenant flag is used. The cache key for AWS creds in this case will also be based on the host and tenant name. In other words, if the tenant ID is provided, the creds will still be stored and retrieved with the tenant name.

## Type of change
- [x] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [DUPLO-9973](https://app.clickup.com/t/8655600/DUPLO-9973) - Some users report cache issues.

### CODE REVIEW

**Comment by duplodavid** [comment]
Before merging, this branch will be retargeted to the develop branch after https://github.com/duplocloud/duplo-jit/pull/21 merges.

**Review by joek-duplo — APPROVED** [review]

**Comment by duplodavid** [comment]
@joek-duplo I just made a merge commit to resolve with the squashed merge commit that went into the develop branch. There are no code changes in that commit, but it needs your sign-off again: https://github.com/duplocloud/duplo-jit/pull/22/commits/ccebfddd77a566710b42e0a977023d7bf9727e2b

Thank you!

**Review by joek-duplo — APPROVED** [review]

**Pull request merged**
